src/App.tsx

```diff
@@ -1,8 +1,74 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { getTodos, getUsers } from './helpers/api';
+import TodoList from './components/TodoList';
+import TodoSort from './components/TodoSort';
 import './App.css';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App = () => {
+  const [todos, setTodos] = useState<Todos[]>([]);
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
+  const loadAllTodos = () => {
+    const todosPromise = getTodos();
+    const usersPromise = getUsers();
+
+    Promise.all([todosPromise, usersPromise])
+      .then(([todosFromServer, usersFromServer]) => {
+        const preparedTodosList = todosFromServer.map(todo => {
+          return {
+            ...todo,
+            user: usersFromServer.find(user => user.id === todo.userId),
+          };
+        });
+
+        setIsLoaded(true);
+        setTodos(preparedTodosList);
+      });
+  };
+
+  const sortByTitle = () => {
+    setTodos([...todos]
+      .sort((a, b) => a.title.localeCompare(b.title)));
+  };
+
+  const sortByStatus = () => {
+    setTodos([...todos]
+      .sort((a, b) => +b.completed - +a.completed));
+  };
+
+  const sortByName = () => {
+    setTodos([...todos]
```

Comment:
you'll lose the initial array if you modify it after each sort;

calculate sorted array before rendering